packages/loader/container-loader/src/containerStorageAdapter.ts

```diff
@@ -120,8 +124,17 @@ export class ContainerStorageAdapter
 			// A callback to ensure we fetch the most updated value of service.policies.summarizeProtocolTree, which could be set
 			// based on the response received from the service after connection is established.
 			() => {
-				this._summarizeProtocolTree =
+				const latestShouldSummarizeProtocolTree =
 					this._summarizeProtocolTree ?? service.policies?.summarizeProtocolTree ?? false;
+				if (this._summarizeProtocolTree !== latestShouldSummarizeProtocolTree) {
+					// log everytime the summarization preference changes
```

Comment:
This comment could be a little misleading - It looks like we only switch the value of `_summarizeProtocolTree` if it's not yet defined.  Once it's set it's fixed for the lifetime of this class.  Right?

Reply:
Would this code be equivalent? I think it's more explicit about the flow:

```
if (this._summarizeProtocolTree === undefined) {
  // ... set it and log
}
return this._summarizeProtocolTree;
```

Reply:
The value can change for _summarizeProtocolTree, whenever there is a reconnection. The code comments for the function aove explain the scenario. Would prefer keeping the comment the same.
Basically, I wish to log every time the value changes.

Reply:
I think @markfields is correct @pragya91 . As per your logic here:
const latestShouldSummarizeProtocolTree =
					this._summarizeProtocolTree ?? service.policies?.summarizeProtocolTree ?? false;

Once this.__summarizeProtocolTree is set (true or false), service.policies?.summarizeProtocolTree won't override it.
So, previously Feature gate and loader option were given preference over service policies as service had nothing to do with differentiation between single/dual commit summary. Not now AFR service is going to decide whether client should make single commit summary or not.
I think you need a different variable here.
If the props taken in the constructor is false, then we cannot override it no matter if the policy says it is true.
However, if that prop is true/undefined, then we respect the service policies.

Reply:
@jatgarg Could you give me more context on why loader setting is to be given preference over the service policy? 

The latest commit I made to enable single-commit for FRS is under the understanding that the service can choose to change the policy on reconnection. It ideally it is supposed to get reset based on the latest values.

Reply:
If it is the host app, I think it is still valid because we want to respect customer's preferences.

Reply:
It is the host app that will set loader policies. 
Problem is that, if the service says that it is not enabled through policies, but it is true in loader options or it was initially set by policies, then this code will not respect the new policy which is problematic. If the service tells that they don't support single commit summary, then we have to respect that otherwise it will cause issues.

Reply:
So do you recommend adding an || instead or ??  @jatgarg

Reply:
@pragya91 - this logic is obviously super subtle with multiple inputs - I would strongly suggest you ensure there is focused unit test coverage of the matrix of various config sources.  Here's an example of how we tested a similar thing for GC:  https://github.com/microsoft/FluidFramework/blob/f21d62b217a128cdc1669431beb4af044c450a2f/packages/runtime/container-runtime/src/test/gc/gcConfigs.spec.ts#L717

Reply:
@markfields  That is an interesting idea, however, for this particular change I am not inclined on adding tests more tests than the ones we already have - they cover both scenarios.